src/backedges.jl

```diff
@@ -105,17 +104,17 @@ function callstring(io, sfs::Vector{StackTraces.StackFrame})
     sf = sfs[end]
     return callstring(io, instance(sfs)) * string(" at ", sf.file, ':', sf.line)
 end
-callstring(io, ipframes::Vector{IPFrames}) = isempty(ipframes) ? "" : callstring(io, ipframes[1].sfs)
+callstring(io::IO, ipframes::Vector{IPFrames}) = isempty(ipframes) ? "" : callstring(io, ipframes[1].sfs)
 
-struct Data{T}
+struct Data
     callstr::String
-    nd::T
+    nd::MethodInstance
```

Comment:
Can't do this, it will break SnoopCompile. I think that's true of most of the changes in this file. I'll see about submitting a "stub" version of what's in SnoopCompile to protect this properly (sorry, it was bad of me not to do that from the outset).

Reply:
ops, thanks. I will revert this part then.

Reply:
We could also just add some integration tests like e.g. in https://github.com/JuliaDiff/ChainRules.jl/blob/master/.github/workflows/IntegrationTest.yml

Reply:
I am all for integration tests!

Reply:
The integration test is nice, thanks so much @simeonschaub !